### ISSUE TITLE
Autocompletion with cite does not work

### ISSUE THREAD

**Issue by jarrndewitt** [vim-latex/vim-latex#113]
I use the following two files and try to get autocompletion of the `\cite` command to work:
`test.tex`
> \documentclass{article}
> \usepackage{natbib}
> \bibliographystyle{dinat}
> \begin{document}
> \cite{s}
> \bibliography{test}
> \end{document}

`test.bib`
> @article{six,
>   title        = "The Six -- a Boring Number",
>   author       = "Name, Alice and Other, Bob",
>   journal      = "Journal for Testing Affairs",
>   year         = 2066,
>   volume       = 66,
>   number       = 6,
>   pages        = "66--666",
> }
> @article{seven,
>   title        = "The Seven -- an Interesting Number",
>   author       = "Name, Alice and Other, Bob",
>   journal      = "Journal for Testing Affairs",
>   year         = 2077,
>   volume       = 77,
>   number       = 7,
>   pages        = "77--777",
> }

When I put the cursor behind the `s` and press `F9` I get two new editor panes.

The first titled `test.tex [+]` containing
> test.bib|1 col 1| @article{six,
> test.bib|10 col 1| @article{seven,
what appears to be some sort of grep results.

The second editor panes is titled `[QuickFix-Liste] :grep.exe -nH ____HIGHLY_IMPROBABLE____ test.tex` and contains the contents of the file test.bib.
Within these editor panes no filtering or navigation as described in the manual is possible.

I enabled debug mode an the log file reads
> view : Tex_Complete: s:type = cite, typeoption = 
> view : :Tex_Complete: using s:prefix = s
> view : +Tex_ScanFileForCite: searching for bibkeys in test.tex (buffer #1)
> view : Tex_ScanFileForCite: found bibliography command in test.tex
> view : trying to search through [test]
> view : finding .bib file [test.tex]
> main : SetTeXOptions: sourcing maps
> comp : +Tex_SetTeXCompilerTarget: setting target to [pdf] for Compiler
> comp : +Tex_SetTeXCompilerTarget: setting target to [pdf] for Viewr
> comp : compiler/tex.vim: sourcing this file
> view : +Tex_SetupCWindow
> view : +Tex_SyncPreviewWindow
> view : :Tex_CompleteRefCiteCustom: completing with 
> view : :Tex_SwitchToInsertMode: 

I use a current vim-latex (downloaded from git) with vim 8.0 (32 bit) on windows 10 (64 bit) and the grep program recommended in the FAQ. Vim-latex loads correctly and everything beside `\cite` completion seems to work.

[I also asked for help on this issue on https://tex.stackexchange.com/questions/404030/vim-latexsuite-bibliography-completion-does-not-work but got no repsonses by now.]

**Comment by gerw** [comment]
The manual http://vim-latex.sourceforge.net/documentation/latex-suite/latex-completion-cite.html describes the cite-completion using python.

Let us check, why your cite completion does not use the python version. What does
````
:echo Tex_HasPython
:echo Tex_UsePython()
:echo Tex_GetVarValue('Tex_UsePython')
:echo Tex_GetVarValue('Tex_UseCiteCompletionVer2')
````
yield?

Do I get it correctly that pressing enter does not complete the \cite command? This should be fixed if you use the latest version from this git repository (from your log it is apparent that you use an old version).

**Comment by jarrndewitt** [comment]
I get

> :echo Tex_HasPython
> 0
> :echo Tex_UsePython()
> 0
> :echo Tex_GetVarValue('Tex_UsePython')
> 1
> :echo Tex_GetVarValue('Tex_UseCiteCompletionVer2')
> 1

Concerning the log: You are right, I pasted the log of an older version. With the current version from git I get
> view : Tex_Complete: s:type = cite, typeoption = 
> view : :Tex_Complete: using s:prefix = s
> view : +Tex_ScanFileForCite: searching for bibkeys.
> view : :Tex_FindBibFiles: 
> view : Tex_FindBibFiles: found bibliography command in test.tex
> view : :Tex_FindBibFiles: bib command from line 10 to line 10
> view : :Tex_FindBibFiles: concatenated bib command: "test}"
> view : :Tex_FindBibFiles: trying to search through [test]
> view : :Tex_FindBibFiles: returning [C:/Users/User/Documents/Temp/test.bib
> ]
> view : finding .bib file [test.tex]
> view : +Tex_SetupCWindow
> view : +Tex_SyncPreviewWindow
> view : :Tex_CompleteRefCiteCustom: completing with six
> view : :Tex_SwitchToInsertMode:

If I press [F9] and then immediately [Enter] the `\cite{s}` becomes `\cite{six}`.

**Comment by gerw** [comment]
Nice that the completion works.

Now about your python support: Does your vim has python?
````
:echo has('python')
:echo has('python3')
````

**Comment by jarrndewitt** [comment]
> :echo has('python')
> 0
> :echo has('python3')
> 0

Although when I do `:version` I see `+python/dyn` as well as `+python3/dyn`. I have Python 2.7.14 for windows installed and working (e.g. at the command line when calling with `python`).

**Comment by gerw** [comment]
The python support in your vim does not work, see https://superuser.com/questions/242938/vim-dyn-support.
If it works, completion should be fine.

**Comment by jarrndewitt** [comment]
Thank you for your help. I actually had a 64bit Python 2.7 and a 32bit vim. When entering e.g. `:python print("A")` I got an error with `E370: Could not load library python27.dll` (as in [https://superuser.com/questions/322138/vim-could-not-load-library-python27-dll#747043](https://superuser.com/questions/322138/vim-could-not-load-library-python27-dll#747043)).

I now installed a 64bit vim and it works.
